Automattic/studio
# Remove event listeners after exporting/importing a site

Related to https://github.com/Automattic/studio/pull/370#discussion_r1689539008.

## Proposed Changes

- Return a function in `handleEvents` to remove all listeners created.
- Remove listeners in the various calls to `handleEvents` in import/export processes.

## Testing Instructions



- Apply the following patch:
```patch
diff --git forkSrcPrefix/src/lib/import-export/handle-events.ts forkDstPrefix/src/lib/import-export/handle-events.ts
index 33633d3d5933ca19517922c812fef68735c75dbe..913a4ad94a32f6419e34cabb206e4bb0d8f115d9 100644
--- forkSrcPrefix/src/lib/import-export/handle-events.ts
+++ forkDstPrefix/src/lib/import-export/handle-events.ts
@@ -23,7 +23,10 @@ export const handleEvents = (
 			} );
 		};
 		emitter.on?.( eventName, listener );
-		removeListeners.push( () => emitter.off?.( eventName, listener ) );
+		removeListeners.push( () => {
+			emitter.off?.( eventName, listener );
+			console.log( 'Listener removed for event', eventName );
+		} );
 	} );
 	return () => {
 		removeListeners.forEach( ( remove ) => remove() );

```
- Follow instructions from https://github.com/Automattic/studio/pull/370.
- After importing a site, observe logs with the message `Removing listeners`.
- After exporting a site, observe logs with the message `Removing listeners`.

## Pre-merge Checklist
